ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -343,13 +351,28 @@ protected String getUuid() {
   public byte[] toBytes() {
     ByteBuffer idBuf = ByteBuffer.allocate(sizeInBytes());
     idBuf.putShort(version);
+    byte flag;
     switch (version) {
       case BLOB_ID_V1:
         break;
       case BLOB_ID_V2:
+        flag = (byte) (type.ordinal() & BLOB_ID_TYPE_MASK);
+        flag |= isEncrypted ? IS_ENCRYPTED_MASK : 0;
+        idBuf.put(flag);
+        idBuf.put(datacenterId);
+        idBuf.putShort(accountId);
+        idBuf.putShort(containerId);
+        break;
       case BLOB_ID_V3:
+        flag = (byte) (type.ordinal() & BLOB_ID_TYPE_MASK);
+        // BLOB_ID_V3 always ignores isEncrypted
```

Comment:
deserialization code in `BlobIdPreamble` needs to be changed. 

Why aren't any tests failing? It looks like there is a gap in the testing then

Reply:
Do you mean remove V3 `isEncryption` parsing in  `BlobIdPreamble` constructor?

Reply:
`testBlobIdFlag()` covers this case and fixed by something like `version >= BLOB_ID_V4 ? isEncrypted : false`

Reply:
yeah - i guess tests won't fail because the flag is `false` during the ser